processor/src/chiplets/ace/mod.rs

```diff
@@ -0,0 +1,345 @@
+#![allow(dead_code)]
+
+use alloc::{collections::BTreeMap, vec::Vec};
+
+use miden_air::{
+    RowIndex,
+    trace::{chiplets::ace::ACE_CHIPLET_NUM_COLS, main_trace::MainTrace},
+};
+use vm_core::{FieldElement, ZERO, mast::BasicBlockNode};
+
+use crate::{
+    ContextId, ExecutionError, Felt, QuadFelt,
+    chiplets::{ace::trace::CircuitEvaluation, memory::Memory},
+    errors::{AceError, ErrorContext},
+    trace::TraceFragment,
+};
+
+mod trace;
+pub use trace::{NUM_ACE_LOGUP_FRACTIONS_EVAL, NUM_ACE_LOGUP_FRACTIONS_READ};
+
+mod instruction;
+#[cfg(test)]
+mod tests;
+
+const PTR_OFFSET_ELEM: Felt = Felt::ONE;
+const PTR_OFFSET_WORD: Felt = Felt::new(4);
+
+/// Arithmetic circuit evaluation (ACE) chiplet.
+///
+/// This is a VM chiplet used to evaluate arithmetic circuits given some input, which is equivalent
+/// to evaluating some multi-variate polynomial at a tuple representing the input.
+///
+/// During the course of the VM execution, we keep track of all calls to the ACE chiplet in an
+/// `EvaluationContext` per call. This is then used to generate the full trace of the ACE chiplet.
```

Comment:
```suggestion
/// [`CircuitEvaluation`] per call. This is then used to generate the full trace of the ACE chiplet.
```

`EvaluationContext` no longer exists